ldes-server-infra-mongo/mongock-changeset-1/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/infra/mongo/mongock/changeset1/MemberUpdaterChange.java

```diff
@@ -0,0 +1,44 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock.changeset1;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.member.entity.LdesMemberEntity;
+import io.mongock.api.annotations.ChangeUnit;
+import io.mongock.api.annotations.Execution;
+import io.mongock.api.annotations.RollbackExecution;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+
+import java.util.List;
+
+@ChangeUnit(id = "member-updater-changeset-1", order = "1", author = "VSDS")
+public class MemberUpdaterChange {
+	private static final String MEMBERS = "members";
+	private final MongoTemplate mongoTemplate;
+
+	public MemberUpdaterChange(MongoTemplate mongoTemplate) {
+		this.mongoTemplate = mongoTemplate;
+	}
+
+	/**
+	 * This is the method with the migration code
+	 **/
+	@Execution
+	public void changeSet() {
+		List<LdesMemberEntityV1> ldesMemberEntityV1s = mongoTemplate.find(new Query(), LdesMemberEntityV1.class);
+		ldesMemberEntityV1s.forEach(ldesMember -> {
+			Query query = new Query();
+			query.addCriteria(Criteria.where(MEMBERS).is(ldesMember.getId()));
+			List<String> treeNodeReferences = mongoTemplate.find(query, LdesFragmentEntityV1.class).stream()
+					.map(LdesFragmentEntityV1::getId).toList();
+			mongoTemplate
+					.save(new LdesMemberEntity(ldesMember.getId(), ldesMember.getLdesMember(), treeNodeReferences));
```

Comment:
I wonder if using the 'real' entities from the codebase can give us problems in the future. Currently we have our V1 entities and the actual codebase is V2 so this matches perfectly. However if larger changes are made in the future, we will have to:
- come back to old migration code to update this code
- changeset2 expects to start from the result of changeset1, if we update this to the latest entities, changeset2 will have a mismatch starting the migration. Maybe we need V2 objects as well? These can then be copy-pasted and become the start of changeset2, etc.?

Reply:
Same for the fragmentUpdater